s3sync_test.go

```diff
@@ -207,6 +208,72 @@ func TestS3sync(t *testing.T) {
 			t.Error("Unexpected keys", objs)
 		}
 	})
+
+	t.Run("UploadEscaped", func(t *testing.T) {
+		// https://github.com/localstack/localstack/issues/8174
+		t.Skip("Skip till localstack would support url escaped characters for key names")
+		temp, err := ioutil.TempDir("", "s3synctest")
+		defer os.RemoveAll(temp)
+
+		escapedFileName := url.QueryEscape("READ%2FME.md")
+
+		if err != nil {
+			t.Fatal("Failed to create temp dir")
+		}
+
+		for _, dir := range []string{
+			filepath.Join(temp, "foo%2Fescape"),
+			filepath.Join(temp, "bar", "baz%2Fescape"),
+			filepath.Join(temp, "pre%2Ffix/foo%2Fescape"),
+			filepath.Join(temp, "pre%2Ffix/bar", "baz%2Fescape"),
```

Comment:
Shouldn't those paths also be `url.QueryEscape`-ed?

Reply:
As I currently can see my initial intention it should be exactly combination of escaped `%2F` as dir names and raw `/` to "separate" the latest. 

So that  it might sync exactly initial `fs` tree state as is without any encoding/transformation.

While `url.QueryEscape` would encode  all  `/` at once.

Still once again I need some time to restore all the details to go on

Reply:
Ah, as we have `escapedFileName` escaped twice it should be escaped in deed.
I will check all once again.